core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -131,6 +132,38 @@ public GetFeatureSetResponse getFeatureSet(GetFeatureSetRequest request)
     return GetFeatureSetResponse.newBuilder().setFeatureSet(featureSet.toProto()).build();
   }
 
+  /**
+   * Finds &amp; returns the featuresets matching the given feature set reference. TODO: merge with
```

Comment:
Yeah, you should merge this method. Add a private method to the jobservice to convert the feature reference to a filter if necessary.

Reply:
@woop wanted the method out of `JobService`, so I moved it to `SpecService`:
> This should almost certainly be externalized. Can you have a look in the code base to see where else it occurs and see if we can reuse or generalize this?
Where should the method be?

Reply:
You can use the existing method in specservice, just add a helper method to convert the `FeatureReference` to `Filter` since the former is a subset of the latter anyway.